src/extensions/timeseries/TimeseriesViewNew/useTimeseriesModel.ts

```diff
@@ -1,229 +1,161 @@
-import { useContext, useRef } from "react"
-import { createCalculationPool, HitherContext } from "../../common/hither"
+import { useContext, useMemo } from "react"
+import { createCalculationPool, HitherContext, HitherInterface } from "../../common/hither"
+import useFetchCache from "../../common/useFetchCache"
 import { RecordingInfo } from "../../extensionInterface"
 import Mda from "./Mda"
 
 interface MdaInterface {
   value: (i1: number, i2: number) => number
 }
 
-class TimeseriesModelNew {
-  _dataSegments = new Map<string, MdaInterface>()
-  _requestDataSegmentHandlers: ((ds_factor: number, segment_num: number) => void)[] = []
-  _dataSegmentSetHandlers: ((ds_factor: number, t1: number, t2: number) => void)[] = []
-  _dataSegmentsRequested = new Map<string, boolean>()
-  constructor(private args: { samplerate: number, num_channels: number, num_timepoints: number, segment_size: number }) {
-  }
-  clear() {
-    this._dataSegments.clear()
-    this._dataSegmentsRequested.clear()
-  }
-  setDataSegment(ds_factor: number, segment_num: number, X: MdaInterface) {
-    const code = `${ds_factor}:${segment_num}`
-    this._dataSegments.set(code, X)
-    this._dataSegmentSetHandlers.forEach(handler => {
-      handler(ds_factor, this.args.segment_size * ds_factor * segment_num, this.args.segment_size * ds_factor * (segment_num + 1))
-    })
-  }
-  onRequestDataSegment(handler: (ds_factor: number, segment_num: number) => void) {
-    this._requestDataSegmentHandlers.push(handler)
-  }
-  onDataSegmentSet(handler: (ds_factor: number, t1: number, t2: number) => void) {
-    this._dataSegmentSetHandlers.push(handler)
+// it may be important to limit this to 1 at a time when using a filter
+const timeseriesCalculationPool = createCalculationPool({maxSimultaneous: 1, method: 'stack'})
+
+export type TimeseriesData = {
+  getChannelData: (ch: number, t1: number, t2: number, ds_factor: number) => number[]
+  requestChannelData: (ch: number, t1: number, t2: number, ds_factor: number) => void
+  numChannels: () => number
+  numTimepoints: () => number
+  getSampleRate: () => number
+}
+
+const getTimeseriesDataSegment = async (args: {hither: HitherInterface, recordingObject: any, ds_factor: number, segment_num: number, segment_size: number}): Promise<Mda> => {
```

Comment:
Sorry, what's `Mda` in this context?

Reply:
Mda stands for multidimensional array. It's an old format I used for holding raw binary data with a small header containing the data type and the array dimensions. We can migrate way from it.